### PR TITLE
Bugfix for EXDATE handling

### PR DESCRIPTION
**Pull request by danesparza**
I wasn't expecting to make a PR request, but here we are.   Thanks for this great repo.  

I discovered a strange edge case with `EXDATE` handling in the library, where an `EXDATE` didn't get used on a recurring event even though it was set.  I couldn't get it to repro on my machine in any unit test, but for some reason I could duplicate the issue in unit tests run on CircleCI and could duplicate the issue when I tried to use the code in AWS Lambda.  

Here is one of the `EXDATE`s in question: `EXDATE;TZID=America/New_York:20220220T173000`

It turns out it was related to timezone!  I was able to see that even though the timezone was present on the line with the EXDATE, gocal was using UTC (or LocalTime, depending on the situation) instead.

So **this pull request is related to parsing and passing EXDATE timezone information** (it was already parsed -- it just needed to be passed along as 'params').  

I also noticed a few issues with the `Test_ReccuringRule` test data as well (I used the [iCalendar validator](https://icalendar.org/validator.html) to spot them): 

- Needed a PRODID and VERSION element
- Events needed to have a unique UID (they didn't)
- The `EXDATE`s had `VALUE=DATE` set, but then also included a time component.  The parser didn't like that.  I updated to use `VALUE=DATE-TIME`

I think that's it.  Let me know if you have comments or concerns.

### CODE REVIEW

File: gocal.go [hunk 1]
```diff
@@ -257,7 +257,11 @@ func (gc *Gocal) parseEvent(l *Line) error {
 
 		gc.buffer.RecurrenceID = l.Value
 	case "EXDATE":
-		d, err := parser.ParseTime(l.Value, map[string]string{}, parser.TimeStart, false, gc.AllDayEventsTZ)
+		/*
+			Reference: https://icalendar.org/iCalendar-RFC-5545/3-8-5-1-exception-date-times.html
+			Several parameters are allowed.  We should pass parameters we have
+		*/
+		d, err := parser.ParseTime(l.Value, l.Params, parser.TimeStart, false, gc.AllDayEventsTZ)
```
Comment (danesparza) [inline gocal.go:264]: Just had to pass in l.Params (vs an empty map)
